x-pack/test/alerting_api_integration/common/plugins/actions_simulators/server/inference_simulation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import http from 'http';
+
+import { ProxyArgs, Simulator } from './simulator';
+
+export class InferenceSimulator extends Simulator {
+  private readonly returnError: boolean;
+
+  constructor({ returnError = false, proxy }: { returnError?: boolean; proxy?: ProxyArgs }) {
```

Comment:
For some of our simulators, we create a single simulator, and determine how to react based on the request coming in.  Maybe a field is set to `returnError` or such, then it returns error instead of the response.  May not work for this case, but thought I'd mention it.